visidata/features/join.py

```diff
@@ -137,7 +137,8 @@ def calcValue(self, row):
             if row[c.sheet] is not None:
                 vals.add(c.getTypedValue(row[c.sheet]))
         if len(vals) != 1:
-            vd.warning(f'inconsistent keys: ' + str(vals))
+            keycolnames = ', '.join([f'{col.sheet.name}:{col.name}' for col in self.keycols])
+            vd.warning(f"The source key columns ({keycolnames}) have different types. Type the source columns before joining.")
```

Comment:
Let's tighten this up.  In general internal messages don't have "the"s or repeated words (like "source columns" and "type").  Plus the user should already know they're joining. This is just a warning and the join will still happen (though it might be incorrect), so the imperative is unclear (is it necessary or not?).

Reply:
This makes a lot of sense to me. I shortened the warning, and it does feel better, and sufficiently informative.